yt_dlp/extractor/tumblr.py

```diff
@@ -107,116 +237,176 @@ class TumblrIE(InfoExtractor):
         'add_ie': ['Instagram'],
     }]
 
+    _providers = {
+        'instagram': 'Instagram',
+        'vimeo': 'Vimeo',
+        'vine': 'Vine',
+        'youtube': 'Youtube',
+    }
+
     def _real_initialize(self):
+        self._ACCESS_TOKEN = None
+        self.get_access_token()
         self._login()
 
+    def get_access_token(self):
+        try:
+            login_page = self._download_webpage(
+                self._LOGIN_URL, None, 'Downloading login page')
+            self._ACCESS_TOKEN = self._search_regex(
+                r'"API_TOKEN":\s*"(\w+)"', login_page, 'API access token')
+        except ExtractorError:
+            self.report_warning('Failed to get access token; metadata will be missing and some videos may not work')
+
     def _login(self):
         username, password = self._get_login_info()
-        if username is None:
+        if not username:
             return
 
-        login_page = self._download_webpage(
-            self._LOGIN_URL, None, 'Downloading login page')
-
-        login_form = self._hidden_inputs(login_page)
-        login_form.update({
-            'user[email]': username,
-            'user[password]': password
-        })
-
-        response, urlh = self._download_webpage_handle(
-            self._LOGIN_URL, None, 'Logging in',
-            data=urlencode_postdata(login_form), headers={
-                'Content-Type': 'application/x-www-form-urlencoded',
-                'Referer': self._LOGIN_URL,
-            })
-
-        # Successful login
-        if '/dashboard' in urlh.geturl():
+        if not self._ACCESS_TOKEN:
+            self.report_warning('Cannot log in without access token')
             return
 
-        login_errors = self._parse_json(
-            self._search_regex(
-                r'RegistrationForm\.errors\s*=\s*(\[.+?\])\s*;', response,
-                'login errors', default='[]'),
-            None, fatal=False)
-        if login_errors:
-            raise ExtractorError(
-                'Unable to login: %s' % login_errors[0], expected=True)
-
-        self.report_warning('Login has probably failed')
+        try:
+            self._download_json(
+                self._OAUTH_URL, None, 'Logging in',
+                data=urlencode_postdata({
+                    'password': password,
+                    'grant_type': 'password',
+                    'username': username,
+                }), headers={
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                    'Authorization': f'Bearer {self._ACCESS_TOKEN}',
+                })
+        except ExtractorError:
+            self.report_warning('Login failed')
 
     def _real_extract(self, url):
         m_url = self._match_valid_url(url)
         video_id = m_url.group('id')
         blog = m_url.group('blog_name')
```

Comment:
compress this into 1 line.
No need to assign match object to var and then call group individually.
Just call groups on the match object